lib/engines/hcktest/tests.rb

```diff
@@ -132,11 +132,36 @@ def wait_queued_test(id)
       end
     end
 
-    def test_parameters(test_name)
+    def select_test_config(test_name, config)
       @project.engine.config.tests_config
               .select { _1.tests.include?(test_name) }
-              .flat_map(&:parameters)
-              .to_h { |parameter| [parameter.name, parameter.value] }
+              .flat_map(&config)
+    end
+
+    def test_parameters(test_name)
+      select_test_config(test_name, :parameters)
+        .to_h { |parameter| [parameter.name, parameter.value] }
+    end
+
+    def run_command_on_client(client, command, desc)
+      @logger.info("Running command (#{desc}) on client #{client.name}")
+      @tools.run_on_machine(client.name, desc, command)
+    end
+
+    def run_test_commands(test, type)
+      select_test_config(test['name'], type).each do |commands|
```

Comment:
You name variables holding `CommandInfo` as "command" instead of "commands". Keep them consistent.